src/components/LoginMenu.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <v-menu
+    v-model="menu"
+    offset-x
+  >
+    <template v-slot:activator="{ on }">
+      <v-btn
+        icon
+        v-on="on"
+      >
+        <v-avatar
+          color="grey lighten-4"
+          size="36px"
+        >
+          <span v-if="userInfo">{{ userInitials }}</span>
+          <v-icon
+            v-else
+            color="grey"
+          >
+            account_circle
+          </v-icon>
+        </v-avatar>
+      </v-btn>
+    </template>
+
+    <v-card>
+      <v-list>
+        <v-list-item class="sign-in-item">
+          <v-list-item-action>
+            <v-btn
+              v-if="userInfo"
+              dark
+              @click="logout"
+            >
+              Log out
+            </v-btn>
+
+            <v-btn
+              v-else
+              class="google-sign-in"
+              dark
+              :ripple="false"
+              :href="loginLink"
+            >
+              <span class="google-logo">
+                <img
+                  alt="Google"
+                  src="../assets/google_badge.svg"
+                >
+              </span>
+              <span class="button-text">
+                Sign in with Google
+              </span>
+            </v-btn>
+          </v-list-item-action>
+        </v-list-item>
+      </v-list>
+    </v-card>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import { host } from '@/environment';
+import store from '@/store';
+import { UserInfo } from '@/types';
+
+export default {
+  data: () => ({
+    menu: false,
+    location: '',
+  }),
+
+  computed: {
+    userInfo: (): UserInfo | null => store.state.userInfo,
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    loginLink(this: any): string {
+      const {
+        location,
+      } = this;
+
+      return `${host}/user/oauth/google/login?return_url=${location}`;
+    },
+
+    userInitials(): string {
+      const {
+        userInfo,
+      } = this;
+
+      if (userInfo !== null) {
+        return `${(userInfo as unknown as UserInfo).given_name[0]}${(userInfo as unknown as UserInfo).family_name[0]}`;
```

Comment:
This `as unknown as UserInfo` stuff is a workaround for poor Vue 2.0 typing in TS. The problem is that technically, `this.userInfo` is a *function* from no arguments to `UserInfo | null`, even though Vue magic allows us (indeed, forces us) to use this as a *value* of that type. So, the type coercion is correct in semantic terms, though obviously it's ugly and reduces the typesafety of the code in general.

When we switch to composition API, this will be one of many TS issues that gets solved.